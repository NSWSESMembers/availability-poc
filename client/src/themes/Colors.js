const Colors = {
  txtMain: '#000',
  txtDark: '#6B6B6B',
  txtWhite: '#ffffff',
  txtError: '#FF5A5F',
  txtSuccess: '#25ce66',
  txtLink: '#007AFF',

  bgMain: '#1e3c78',
  bgBtn: '#f48603',
  bgBtnSecondary: '#1c87c9',
  bgBtnAvailable: 'green',
  bgBtnUnavailable: '#FF5A5F',
  bgBtnUrgent: 'purple',
  bgOutOfRange: '#AAA',
  bgLight: '#EEE',
  bgWhite: '#ffffff',
  bgDisabled: '#D3D3D3',
  bgError: '#FF5A5F',
  bgChat: '#f1f1f1',

  bdMain: '#00a699',
  bdWhite: '#ffffff',
  bdDisabled: '#D3D3D3',
  bdLine: '#dddddd',
  bdInput: '#cbcbcb',
  bdLight: '#DDD',
};

export default Colors;
