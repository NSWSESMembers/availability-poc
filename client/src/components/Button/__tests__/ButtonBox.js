import React from 'react';
import renderer from 'react-test-renderer';

import { ButtonBox } from '../';
import Colors from '../../../themes/Colors';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <ButtonBox
      text="text"
      subtext="subtext"
      onPress={noop}
      disabled
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <ButtonBox
      text="text"
      subtext="subtext"
      onPress={noop}
      selected
      selectedColor={Colors.bgBtnAvailable}
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();

  const tree3 = renderer.create(
    <ButtonBox
      text="text"
      onPress={noop}
      selectedColor={Colors.bgBtnUnavailable}
    />,
  ).toJSON();
  expect(tree3).toMatchSnapshot();

  const tree4 = renderer.create(
    <ButtonBox
      text="text"
      onPress={noop}
      selectedColor={Colors.bgBtnUrgent}
    />,
  ).toJSON();
  expect(tree4).toMatchSnapshot();
});
