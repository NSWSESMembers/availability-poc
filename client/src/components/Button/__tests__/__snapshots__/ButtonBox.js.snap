// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderWidth": 1,
        "padding": 10,
      },
      Object {
        "backgroundColor": "white",
        "borderColor": "#AAA",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "color": "#AAA",
        },
      ]
    }
  >
    text
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#AAA",
        },
      ]
    }
  >
    subtext
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 18,
        },
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "white",
          },
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "green",
      "borderColor": "green",
      "borderWidth": 1,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    text
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    subtext
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 18,
        },
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "white",
          },
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`renders correctly 3`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderWidth": 1,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
  >
    text
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 18,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 18,
        },
        Array [
          Object {
            "textAlign": "center",
          },
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`renders correctly 4`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderWidth": 1,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
  >
    text
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 18,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 18,
        },
        Array [
          Object {
            "textAlign": "center",
          },
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
